ksql-api/src/main/java/io/confluent/ksql/api/server/ConnectionQueryManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpConnection;
+import io.vertx.core.http.HttpServerRequest;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class ConnectionQueryManager {
+
+  private final Context context;
+  private final Server server;
+  private final Map<HttpConnection, ConnectionQueries> connectionsMap = new HashMap<>();
+
+  public ConnectionQueryManager(final Context context, final Server server) {
+    this.context = Objects.requireNonNull(context);
+    this.server = Objects.requireNonNull(server);
+  }
+
+  public PushQueryHolder createApiQuery(final QuerySubscriber querySubscriber,
+      final HttpServerRequest request) {
+    checkContext();
+    final ConnectionQueries connectionQueries = getConnectionQueries(request);
+    final PushQueryHolder query = new PushQueryHolder(server, querySubscriber,
+        connectionQueries::removeQuery);
+    connectionQueries.addQuery(query);
+    return query;
+  }
+
+  private ConnectionQueries getConnectionQueries(final HttpServerRequest request) {
+    final HttpConnection conn = request.connection();
+    ConnectionQueries connectionQueries = connectionsMap.get(conn);
+    if (connectionQueries == null) {
+      connectionQueries = new ConnectionQueries(conn);
+      connectionsMap.put(conn, connectionQueries);
+    }
+    return connectionQueries;
+  }
+
+  private void checkContext() {
+    if (Vertx.currentContext() != context) {
+      throw new IllegalStateException("On wrong context");
+    }
+  }
```

Comment:
We see this method in a lot of places. Shame Vert.x `Context` doesn't have a method that does this ;) e.g. `context.throwIfDifferentContext()`.

I guess we could have a `VertxUtils` class that had this as a util method. I know its only small, but maybe worth it.